src/main/java/baritone/command/defaults/HauseArea.java

```diff
@@ -0,0 +1,265 @@
+                "The expand/contract/shift commands use a kind of selector to choose which selections to target. Supported ones are a/all, n/newest, and o/oldest.",
+                "",
+                "Usage:",
+                "> ha pos1/p1/1 - Set position 1 to your current position.",
+                "> ha pos1/p1/1 <x> <y> <z> - Set position 1 to a relative position.",
+                "> ha pos2/p2/2 - Set position 2 to your current position.",
+                "> ha pos2/p2/2 <x> <y> <z> - Set position 2 to a relative position.",
+                "",
+                "PS: It's a blatant and shameful #sel copy"
+        );
+    }
+
+    enum Action {
+        POS1("pos1", "p1", "1"),
+        POS2("pos2", "p2", "2"),
+        UNDO("undo", "u"),
+        CONTRACT("contract", "ct"),
+        SHIFT("shift", "sh"),
+        EXPAND("expand", "ex"),
+        CLEAR("clear");
+        private final String[] names;
+
+        Action(String... names) {
+            this.names = names;
+        }
+
+        public static Action getByName(String name) {
+            for (Action action : Action.values()) {
+                for (String alias : action.names) {
+                    if (alias.equalsIgnoreCase(name)) {
+                        return action;
+                    }
+                }
+            }
+            return null;
+        }
+
+        public static String[] getAllNames() {
+            Set<String> names = new HashSet<>();
+            for (Action action : Action.values()) {
+                names.addAll(Arrays.asList(action.names));
+            }
+            return names.toArray(new String[0]);
+        }
+    }
+
+    enum TransformTarget {
+        ALL(sels -> sels, "all", "a"),
+        NEWEST(sels -> new ISelection[]{sels[sels.length - 1]}, "newest", "n"),
+        OLDEST(sels -> new ISelection[]{sels[0]}, "oldest", "o");
+        private final Function<ISelection[], ISelection[]> transform;
+        private final String[] names;
+
+        TransformTarget(Function<ISelection[], ISelection[]> transform, String... names) {
+            this.transform = transform;
+            this.names = names;
+        }
+
+        public ISelection[] transform(ISelection[] selections) {
+            return transform.apply(selections);
+        }
+
+        public static TransformTarget getByName(String name) {
+            for (TransformTarget target : TransformTarget.values()) {
+                for (String alias : target.names) {
+                    if (alias.equalsIgnoreCase(name)) {
+                        return target;
+                    }
+                }
+            }
+            return null;
+        }
+
+        public static String[] getAllNames() {
+            Set<String> names = new HashSet<>();
+            for (TransformTarget target : TransformTarget.values()) {
+                names.addAll(Arrays.asList(target.names));
+            }
+            return names.toArray(new String[0]);
+        }
+    }
+}
```

Comment:
Can the existing sel command be reused?

Reply:
i was planning doing it today, i just wanted to get it running